api/storage_service.cc

```diff
@@ -306,19 +306,19 @@ future<scrub_info> parse_scrub_options(const http_context& ctx, sharded<db::snap
 }
 
 void set_transport_controller(http_context& ctx, routes& r, cql_transport::controller& ctl, seastar::scheduling_group sg) {
-    ss::start_native_transport.set(r, [&ctx, &ctl](std::unique_ptr<http::request> req) {
+    ss::start_native_transport.set(r, [&ctl, sg](std::unique_ptr<http::request> req) {
         return smp::submit_to(0, [&] {
-            return with_scheduling_group(ctx.db.local().get_statement_scheduling_group(), [&ctl] {
+            return with_scheduling_group(sg, [&ctl] {
```

Comment:
Instead of propagating sg here, you could propagate it to the server itself, so it will take care of the switching.
See https://github.com/scylladb/scylladb/issues/17643.

Reply:
What does thumb-up mean here? You will send next version implementing it? Or you just think it is a good idea... for a follow-up... that someone else will do?

Reply:
I will send next version implementing it. I thought of this too, but then judged that it's maybe not worth it. Now when there's an issue with good justification, there's no excuse to off-load it to followup.